src/IrishModel.cpp

```diff
@@ -4,8 +4,8 @@ namespace EERAModel {
 namespace Model {
 
 IrishModel::IrishModel(const ModelInputParameters& modelInputParameters,
-    InputObservations& observations, Random::RNGInterface::Sptr rng, Utilities::logging_stream::Sptr log) 
-    : rng_(rng) {
+    const InputObservations& observations, Random::RNGInterface::Sptr rng, const Utilities::logging_stream::Sptr& log) 
```

Comment:
Same comment as for the `InferenceFramework` constructor, regarding the reference to a shared pointer